driver-core/src/main/com/mongodb/internal/operation/OperationHelper.java

```diff
@@ -143,29 +143,25 @@ static void validateCollationAndWriteConcern(final ConnectionDescription connect
         }
     }
 
-    static void validateArrayFiltersAndWriteConcern(final ConnectionDescription connectionDescription,
-                                                    final List<BsonDocument> arrayFilters, final WriteConcern writeConcern) {
-        if (arrayFilters != null) {
-            if (serverIsLessThanVersionThreeDotSix(connectionDescription)) {
-                throw new IllegalArgumentException(format("Array filters not supported by wire version: %s",
-                        connectionDescription.getMaxWireVersion()));
-            } else if (!writeConcern.isAcknowledged()) {
-                throw new MongoClientException("Specifying array filters with an unacknowledged WriteConcern is not supported");
-            }
+    private static void validateArrayFilters(final ConnectionDescription connectionDescription,
+                                             final List<BsonDocument> arrayFilters, final WriteConcern writeConcern) {
```

Comment:
the arrayFilters argument is no longer used

Reply:
Removed.